server/routes/referral.routes.js

```diff
@@ -139,4 +139,50 @@ router.post(
   ReferralController.sendInvite,
 );
 
+/**
+ * @swagger
+ * /referral/refcode/:refcode:
+ *   get:
+ *     tags:
+ *       - Referral
+ *     description: Get a user's information by refcode
+ *     produces:
+ *       - application/json
+ *     requestBody:
+ *      content:
+ *        application/json:
+ *          schema:
+ *            $ref: '#/components/schemas/Referral'
+ *      description: Get a user's information by refcode
+ *     responses:
+ *      '200':
+ *        description: returns referral
+ *      '500':
+ *       description: Internal server error
+ */
+router.get('/refcode/:refcode', authenticate, ReferralController.getUserByRefCode);
```

Comment:
Doesn't really matter, but I feel `ref` is shorter and convenes the same meaning.
```suggestion
router.get('/ref/:refcode', authenticate, ReferralController.getUserByRefCode);
```

Reply:
Also, get referral by refcode and id should be public endpoint. Referees are unauthenticated so  you shouldn't authenticate before accessing them.